src/views/virtualmachines/tree/hooks/useSyncClicksEffects.ts

```diff
@@ -0,0 +1,52 @@
+import { useEffect } from 'react';
+import { Location } from 'react-router-dom-v5-compat';
+
+import { ALL_NAMESPACES, ALL_NAMESPACES_SESSION_KEY } from '@kubevirt-utils/hooks/constants';
+import { VirtualMachineModelRef } from '@kubevirt-utils/models';
+
+import { FOLDER_SELECTOR_PREFIX, PROJECT_SELECTOR_PREFIX } from '../utils/constants';
+import { setSelectedTreeItem, treeDataMap } from '../utils/utils';
+
+export const useSyncClicksEffects = (
+  activeNamespace: string,
+  loaded: boolean,
+  location: Location<any>,
+) => {
+  useEffect(() => {
+    const queryParams = new URLSearchParams(location.search);
+
+    const folderFilterName = queryParams.get('labels')?.split('=')?.[1];
+    const pathname = location.pathname;
+    if (loaded) {
+      const dataMap = treeDataMap.value;
+      if (pathname.startsWith(`/k8s/${ALL_NAMESPACES}`)) {
+        setSelectedTreeItem(dataMap[ALL_NAMESPACES_SESSION_KEY]);
+        return;
+      }
+
+      if (
+        pathname.startsWith('/k8s/ns/') &&
+        (pathname.endsWith(VirtualMachineModelRef) ||
+          pathname.endsWith(`${VirtualMachineModelRef}/`))
+      ) {
+        if (folderFilterName) {
+          setSelectedTreeItem(
+            dataMap[`${FOLDER_SELECTOR_PREFIX}/${activeNamespace}/${folderFilterName}`],
+          );
+          return;
+        }
+
+        setSelectedTreeItem(dataMap[`${PROJECT_SELECTOR_PREFIX}/${activeNamespace}`]);
+        return;
+      }
+
+      const vmName = pathname.split('/')[5];
+      if (vmName && pathname.includes(vmName)) {
```

Comment:
u are extracting vmNAme from path and checking its there?

Reply:
you're right, changed to just check if the name exists